docs/rfcs/2024-04-01-super-peers-2.md

```diff
@@ -0,0 +1,262 @@
+
+Overall, while repudiation of sent messages appears as important, it seems much less important for administrative changes, and signing member and group changes while relying on Merkle DAG for message history integrity appears to be an optimal tradeoff.
+
+To support this functionality the `admin` and `owner` members have to add public keys to their profiles and communicate profile updates to all members (before or after group is switched to super-peers, but probably before is better).
+
+The middle ground here is moderation, and trade-off here is more nuanced. Practically, whether message is fully removed or marked as removed is the group policy decision, and it can be made based on whether it's more important to preserve content or to remove undesirable content without trace, and also given that super-peers are likely to be give automatic moderation capabilities anyway, preserving deniability for moderation events and relying on Merkle DAG to identify integrity violation seems a better alternative than signing moderation messages. Although it can also be a part of group policy whether to require signing moderation events.
+
+The change to member profile will be:
+
+```haskell
+data Profile = Profile
+  { displayName :: ContactName,
+    fullName :: Text,
+    image :: Maybe ImageData,
+    contactLink :: Maybe ConnReqContact,
+    preferences :: Maybe Preferences,
+    authKey :: Maybe AuthKey -- new field
+  }
+
+-- this is rather ad hoc and tries to allow two things:
+-- - verify that the member has the private key.
+-- - allow key rotations on profile changes, without signing the whole profile change.
+-- A better option could be to use certificates, although they are much large in size,
+-- and also to simply sign profile changes that include key change (then prevKeySignature won't be needed)
+data AuthKey = AuthKey
+  { key :: PublicKeyEd25519,
+    signature :: SignatureEd25519, -- signature of the key itself
+    prevKeySignature :: Maybe SignatureEd25519
+  }
+```
+
+## Role, rank and moderation permission changes of members and group profile changes
+
+There are two problems with the current approach to changing permission, when a single member makes this decision:
+- member whose role changes may disagree. E.g., the member may not be willing to have owner or admin role in some groups. Even more so, the member may be unable to perform super-peer functions (have rank 1).
+- other owners or admins may disagree with the change or it may have been made by mistake.
+
+With the move to super-peers it is additionally complicated by the fact that super-peers will be forwarding these messages, and they could be compromised, thus disrupting group functioning - this risk currently exists with the existing group directory that plays admin role and in case it is compromised it can remove all members from the group.
+
+Part of this problem is addressed in the previous point by requiring to sign administrative changes.
```

Comment:
It seems futile to hope for reliability in group where super-peers are compromised, especially with 2 super-peers and unclear way to reach consensus. I think rfc leaned too early into idea of any members being able to be super-peers. Perhaps it is not required for regular clients to be able to be super peers (or at least via the same member record that can compose messages). Instead owners could add an "invisible" member record to group to act as super-peer (in case they use a highly available client, which usually isn't a case even if it's desktop app, since people turn it off/put to sleep, so it doesn't seem that valuable for most groups), or request "hosted" super-peers (similar to simplex-directory) up to certain quota (min 3 max 5? or self-hosted to differentiate from chat client), or just default to 3 for MVP.